specs/_features/eip7594/polynomial-commitments-sampling.md

```diff
@@ -340,12 +340,26 @@ def verify_kzg_proof_multi_impl(commitment: KZGCommitment,
                                 ys: Sequence[BLSFieldElement],
                                 proof: KZGProof) -> bool:
     """
-    Helper function that verifies a KZG multiproof
+    Verify a KZG multi-evaluation proof for a set of `k` points.
+
+    This is done by checking if the following equation holds:
+        Q(x) Z(x) = f(X) - r(X)
+    Where:
+        f(X) is the polynomial that we want to show opens at `k` points to `k` values
+        Q(X) is the quotient polynomial computed by the prover
+        r(X) is the degree `k-1` polynomial that agrees with f(x) at all `k` points
```

Comment:
"agrees with f(x)" seems like the wrong PoV here, since the verifier does not know if f(x) actually evaluates to the right values at the right points.

Pehraps "r(x) is the degree `k-1` polynomial that evaluates to ``ys`` at all ``zs`` points"? Or something similar.

It seems a bit weird to not use the ``ys`` and ``zs`` inputs anywhere in the doc description.

Finally, what about renaming `r(x)`  to `I(x)` standing for interpolation polynomial? No strong opinion about this one. I know Dankrad used this notation [in his post](https://dankradfeist.de/ethereum/2020/06/16/kate-polynomial-commitments.html) and we also used it [in ours](https://ethresear.ch/t/a-universal-verification-equation-for-data-availability-sampling/13240).

Reply:
Will take your suggestion for the description and rename r(X) to I(X) -- I think r(X) describes what it is, while I(X) describes how it was made (by interpolating the values). In this case, to avoid confusion from folks reading multiple documents, using I(X) is the sound approach